Content.Shared/Kitchen/SharedReagentGrinderComponent.cs

```diff
@@ -0,0 +1,122 @@
+﻿using System;
+using Content.Shared.Chemistry;
+using Content.Shared.GameObjects;
+using Robust.Shared.GameObjects;
+using Robust.Shared.GameObjects.Components.UserInterface;
+using Robust.Shared.Serialization;
+
+namespace Content.Shared.Kitchen
+{
+    public class SharedReagentGrinderComponent : Component
+    {
+        public override string Name => "ReagentGrinder";
+        public override uint? NetID => ContentNetIDs.REAGENT_GRINDER;
+
+
+        [Serializable, NetSerializable]
+        public class ReagentGrinderGrindStartMessage : BoundUserInterfaceMessage
+        {
+            public ReagentGrinderGrindStartMessage()
+            {
+            }
+        }
+
+        [Serializable, NetSerializable]
+        public class ReagentGrinderJuiceStartMessage : BoundUserInterfaceMessage
+        {
+            public ReagentGrinderJuiceStartMessage()
+            {
+            }
+        }
+
+        [Serializable, NetSerializable]
+        public class ReagentGrinderEjectChamberAllMessage : BoundUserInterfaceMessage
+        {
+            public ReagentGrinderEjectChamberAllMessage()
+            {
+            }
+        }
+
+        [Serializable, NetSerializable]
+        public class ReagentGrinderEjectBeakerMessage : BoundUserInterfaceMessage
+        {
+            public ReagentGrinderEjectBeakerMessage()
+            {
+            }
+        }
+
+        [Serializable, NetSerializable]
+        public class ReagentGrinderEjectChamberContentMessage : BoundUserInterfaceMessage
+        {
+
+            public EntityUid EntityID;
+            public ReagentGrinderEjectChamberContentMessage(EntityUid entityID)
+            {
+                EntityID = entityID;
+            }
+        }
+
+        [Serializable, NetSerializable]
+        public class ReagentGrinderVaporizeReagentIndexedMessage : BoundUserInterfaceMessage
+        {
+
+            public Solution.ReagentQuantity ReagentQuantity;
+            public ReagentGrinderVaporizeReagentIndexedMessage(Solution.ReagentQuantity reagentQuantity)
+            {
+                ReagentQuantity = reagentQuantity;
+            }
+        }
+
+        [Serializable, NetSerializable]
+        public class ReagentGrinderWorkStartedMessage : BoundUserInterfaceMessage
+        {
+            public bool IsJuiceIntent;
+
+            public ReagentGrinderWorkStartedMessage(bool wasJuiceIntent)
+            {
+                IsJuiceIntent = wasJuiceIntent;
+            }
+        }
+
+        [Serializable, NetSerializable]
+        public class ReagentGrinderWorkCompleteMessage : BoundUserInterfaceMessage
+        {
+
+            public ReagentGrinderWorkCompleteMessage()
+            {
+            }
+        }
+
+        [Serializable, NetSerializable]
+        public enum ReagentGrinderVisualState
+        {
+            NoBeaker,
+            BeakerAttached
```

Comment:
Could just have the one value and check a bool

Also make it enum ReagentGrinderVisualState : byte for the minor savings (by default they're ints).